pkg/registry/extensions/daemonset/strategy_test.go

```diff
@@ -31,3 +37,67 @@ func TestDefaultGarbageCollectionPolicy(t *testing.T) {
 		t.Errorf("DefaultGarbageCollectionPolicy() = %#v, want %#v", got, want)
 	}
 }
+
+func TestSelectorImmutability(t *testing.T) {
+	tests := []struct {
+		method                 string
+		url                    string
+		oldSelectorLabels      map[string]string
+		newSelectorLabels      map[string]string
+		expectedAPIVersion     string
+		expectedSelectorLabels map[string]string
+	}{
+		{"GET", "/api/v1beta2/namespaces", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "v1beta2", map[string]string{"a": "b"}},
+		{"GET", "/api/v1beta1/namespaces", map[string]string{"a": "b"}, map[string]string{"c": "d"}, "v1beta1", map[string]string{"c": "d"}},
+	}
+
+	resolver := newTestRequestInfoResolver()
+
+	for _, test := range tests {
+		req, _ := http.NewRequest(test.method, test.url, nil)
+
+		apiRequestInfo, err := resolver.NewRequestInfo(req)
+		if err != nil {
+			t.Errorf("Unexpected error for url: %s %v", test.url, err)
+		}
+		if !apiRequestInfo.IsResourceRequest {
```

Comment:
@kow3ns 
After writing the unit test, i think we can leave out the API group when determining whether to ignore changes to `Spec.Selector`. The main reason is due to how `RequestInfo` works.

`IsResourceRequest` returns true for requests to API resources, like `/api/v1/...`, and false for non-resource endpoints like `/api`, `/healthz`, and `/swaggerapi`. In other words, something like `api/v1beta2/...` works, but not `/api/apps/v1beta2/...` (additional `apps` after `api`) or `/apps/v1beta2/...` (do not start with `api`).

Furthermore, we will have the silent ignore for `v1` in future, which will be `/api/v1/...` that works perfectly with how `RequestInfo` works. Moreover, `v1beta2` only exists for `apps` group, which does not justify verifying the API group.

WDYT?